internal/transportmock/matcher.go

```diff
@@ -0,0 +1,59 @@
+package transportmock
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+type Matcher func(req *http.Request) bool
+
+func REST(method, p string) Matcher {
```

Comment:
What do we think about the naming scheme here? Should we prefix/postfix `Match`/`Matcher` to this and the `GQL` function?